### PR TITLE
Fix cdi mode resolution

### PR DESCRIPTION
**Pull request by elezar**
This change incorporates the changes from https://github.com/NVIDIA/go-nvlib/pull/28 to make the mode resolution consistent.

This ensures that CDI spec generation works on Tegra-based systems where nvml is present.

### CODE REVIEW

File: internal/info/auto.go [hunk 1]
```diff
@@ -17,75 +17,40 @@
...
+	nvinfo := info.New(
+		info.WithLogger(logger),
+		info.WithProperties(properties),
+	)
 
-	if (isTegra && !hasNVML) || usesNVGPUModule {
+	switch nvinfo.ResolvePlatform() {
+	case info.PlatformNVML, info.PlatformWSL:
+		return "legacy"
+	case info.PlatformTegra:
```
[the remark this answers is not in the document]
Comment (elezar) [reply internal/info/auto.go:52]: Yes, the logic for checking the use of the `nvgpu` module has been pulled into the platform check. `PlatformTegra` now means that either the Tegra sysfs files are present and NVML is not, or NVML is present and the `nvgpu` module is used.

File: internal/info/auto_test.go [hunk 2]
```diff
@@ -23,6 +23,8 @@ import (
 	testlog "github.com/sirupsen/logrus/hooks/test"
 	"github.com/stretchr/testify/require"
 
+	"github.com/NVIDIA/go-nvlib/pkg/nvlib/info"
+
```
Comment (ArangoGutierrez) [inline internal/info/auto_test.go:27]: is this extra line wanted/needed?

**Review by ArangoGutierrez — COMMENTED** [review]
Just a couple comments, looks good

File: internal/info/auto_test.go (hunk 2 above)
Comment (elezar) [reply internal/info/auto_test.go]: You're right. It isn't.

File: pkg/nvcdi/lib.go [hunk 3]
```diff
@@ -180,30 +185,23 @@ func (m *wrapper) GetCommonEdits() (*cdi.ContainerEdits, error) {
 
 // resolveMode resolves the mode for CDI spec generation based on the current system.
 func (l *nvcdilib) resolveMode() (rmode string) {
-	if l.mode != ModeAuto {
+	if l.mode != "auto" {
```
[the remark this answers is not in the document]
Comment (elezar) [reply pkg/nvcdi/lib.go]: Updated.

**Review by ArangoGutierrez — APPROVED** [review]
Looks good to me

**Pull request merged**